### PR TITLE
chore: upgrade husky to v8 and lint-staged to v13

### PR DESCRIPTION
**Pull request by chimurai**
### What does it do?

upgrade husky to v8 and lint-staged to v13

### Why is it needed?

Getting rid of really old transitive husky@3 dependencies.

Husky v8:
- Zero dependencies and lightweight (6 kB)
- Powered by modern new Git feature (core.hooksPath)
- Follows [npm](https://docs.npmjs.com/cli/v8/using-npm/scripts#best-practices) and [Yarn](https://yarnpkg.com/advanced/lifecycle-scripts#a-note-about-postinstall) best practices regarding autoinstall

### How to test it?

Verified git hooks locally with new husky@8

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     ".github/actions/*"
   ],
   "scripts": {
+    "prepare": "husky install",
```
Comment (chimurai) [inline package.json:31]: > Why husky doesn't autoinstall anymore

https://blog.typicode.com/husky-git-hooks-autoinstall/

File: package.json [hunk 2]
```diff
@@ -84,7 +80,7 @@
     "fs-extra": "10.1.0",
     "get-port": "5.1.1",
     "glob": "7.2.3",
-    "husky": "3.1.0",
+    "husky": "^8.0.1",
```
Comment (joshuaellis) [inline package.json:83]: inline with the rest of the dependencies this should be locked please.

**Review by joshuaellis — CHANGES_REQUESTED** [review]
minor change request, see comment

File: package.json (hunk 2 above)
Comment (chimurai) [reply package.json]: pinned the husky version

**Review by joshuaellis — APPROVED** [review]

**Comment by joshuaellis** [comment]
Thanks for the contribution @chimurai! ⭐

**Pull request merged**